### PR TITLE
fix(Stack): added failed gesture state to clear interaction

### PR DESCRIPTION
**Pull request by Iltimirov**
Added failed gesture state to clear interaction manager queue 

**Motivation**

My team ran into an issue on IOS where the user taps somewhere on the left edge of the screen without any swipe, and upon releasing their finger, the InteractionManager has one outstanding task in the queue. And this task can block user interactions wrapped into `InteractionManager.runAfterInteractions`

It could be related with https://github.com/react-navigation/react-navigation/issues/8606 but i'm not sure

**Test plan**

1. Create a stack with two screen with an ability to navigate by swipe
2. Navigate from screen (1) to screen (2)
3. Tap somewhere in the left edge of the screen
4. After that all actions wrapped into `InteractionManager.runAfterInteractions` will not work

### CODE REVIEW

File: packages/stack/src/views/Stack/Card.tsx [hunk 1]
```diff
@@ -336,6 +336,10 @@ export class Card extends React.Component<Props> {
         onGestureEnd?.();
         break;
       }
+      case GestureState.FAILED: {
+        this.handleEndInteraction();
```
Comment (satya164) [inline packages/stack/src/views/Stack/Card.tsx:340]: I'm not sure if this is enough. Isn't `this.isSwiping.setValue(FALSE);` needed as well?

Comment (Iltimirov) [reply packages/stack/src/views/Stack/Card.tsx:340]: Hmm, maybe. Thank you for review! I'll check it

Comment (Iltimirov) [reply packages/stack/src/views/Stack/Card.tsx:341]: done

**Review by satya164 — APPROVED** [review]
Thank you

**Pull request merged**